src/index.ts

```diff
@@ -49,6 +50,26 @@ declare module "postgraphile" {
      * stores (e.g. S3).
      */
     persistedOperationsGetter?: PersistedOperationGetter;
+
+    /**
+     * There are situations where you may want to allow arbitrary operations
+     * (for example using GraphiQL in development, or allowing an admin to
+     * make arbitrary requests in production) whilst enforcing Persisted
+     * Operations for the application and non-admin users. This function
+     * allows you to determine under which circumstances persisted operations
+     * may be bypassed.
+     *
+     * @example
+     *
+     * ```
+     * app.use(postgraphile(DATABASE_URL, SCHEMAS, {
+     *   allowUnpersistedOperation(req) {
+     *     return process.env.NODE_ENV === "development" && req.headers.referer.endsWith("/graphiql");
+     *   }
+     * });
+     * ```
+     */
+    allowUnpersistedOperation?(request: IncomingMessage, payload: any): boolean;
```

Comment:
As below, we could have this accept a boolean too:

```suggestion
    allowUnpersistedOperation?: boolean | ((request: IncomingMessage, payload: any) => boolean);
```

(This change would need copying to the README also.)